contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -832,20 +832,41 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
    * @notice median from the most recent report
    */
   function latestAnswer() public view virtual override returns (int256) {
-    return s_transmissions[s_hotVars.latestAggregatorRoundId].answer;
+    return s_transmissions[uint32(latestRound())].answer;
   }
 
   /**
    * @notice timestamp of block in which last report was transmitted
    */
   function latestTimestamp() public view virtual override returns (uint256) {
-    return s_transmissions[s_hotVars.latestAggregatorRoundId].recordedTimestamp;
+    return s_transmissions[uint32(latestRound())].recordedTimestamp;
   }
 
   /**
-   * @notice Aggregator round (NOT OCR round) in which last report was transmitted
+   * @notice Aggregator round (NOT OCR round) in which last valid report was transmitted
    */
   function latestRound() public view virtual override returns (uint256) {
+    Transmission memory transmission;
+    
+    // check if the message sender is the secondary proxy
+    if (msg.sender == s_secondaryProxy) {
+      transmission = s_transmissions[s_hotVars.latestSecondaryRoundId];
+      // in case the latest secondary round does not accomplish the cutoff time condition,
+      // get the round id syncing with the primary rounds
+      if (transmission.recordedTimestamp + s_cutoffTime < block.timestamp) {
+        return _getSyncPrimaryRound();
+      }
+
+      // in case the latest secondary round accomplish the cutoff time condition, return it
+      return s_hotVars.latestSecondaryRoundId;
+    }
+    // if the message sender is not the secondary proxy, get the latest primary round id
+    transmission = s_transmissions[s_hotVars.latestAggregatorRoundId];
+    // in case the report was sent in the same block, get the previous round id
+    if (transmission.recordedTimestamp == block.timestamp) {
+      return s_hotVars.latestAggregatorRoundId-1;
+    }
```

Comment:
are we sure this is the right logic? what if the previous round doesn't fall outside of the cutoff time?

Reply:
This lines are for the primary proxy. If the caller is the secondary proxy it'll execute the code inside the if.

For the primary proxy we know that if a recorded transmission is the same as the block timestamp, the previous one will be older and available.